src/tir/schedule/error.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef TVM_TIR_SCHEDULE_ERROR_H_
+#define TVM_TIR_SCHEDULE_ERROR_H_
+
+#include <tvm/tir/schedule/state.h>
+
+namespace tvm {
+namespace tir {
+
+/*! \brief Error that happens during TensorIR scheduling */
+class ScheduleError : public tvm::runtime::Error {
+ public:
+  /*! \brief Base constructor */
+  ScheduleError() : tvm::runtime::Error("") {}
+  /*! \brief The schedule primitive applied that causes the error */
+  virtual String primitive() const = 0;
+  /*! \brief The IRModule that is applied on */
+  virtual IRModule mod() const = 0;
+  /*! \brief The locations of interest that we want to point out */
+  virtual Array<ObjectRef> LocationsOfInterest() const = 0;
+  /*!
+   * \brief Returns an error string template for rendering, corresponds to the "detail" mode.
+   * \sa ScheduleErrorRenderLevel
+   * \note Format of the template:
+   * "Some normal error string and curly bracket-enclosed numbers like {0}, which indicates the
+   * 0-th element of the interested location."
```

Comment:
I'm not sure I understand this sentence.

Reply:
Hmmm would you like to help me rephrase? The template is a string like:

> "Some error occurred on block {0} and loop {1} blablabla"

And renderer will replace {0} and {1} according to the list provided `LocationsOfInterest`. Right now it only printed out all the locations in plain text, but in the future, we may want to mark the IR with underscores and attach names to each location of interest, like what synr does.

Reply:
The explanation you've provided here is good. Can you put it in the comment?

Reply:
Thanks Tristan!